service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/azure/AzureStorageAccessServiceUnitTest.java

```diff
@@ -422,19 +424,128 @@ void createAzureStorageContainerSasToken_blobName() throws InterruptedException
             List.of(
                 SamConstants.SamControlledResourceActions.READ_ACTION,
                 SamConstants.SamControlledResourceActions.WRITE_ACTION));
+    setupMocks(storageAccountResource, storageContainerResource, Optional.empty());
 
     var result =
         azureStorageAccessService.createAzureStorageContainerSasToken(
-            UUID.randomUUID(),
-            storageContainerResource,
-            storageAccountResource,
-            startTime,
-            expiryTime,
+            storageContainerResource.getWorkspaceId(),
+            storageContainerResource.getResourceId(),
             userRequest,
-            null,
-            "foo/the/bar.baz",
-            null);
+            new SasTokenOptions(null, startTime, expiryTime, "foo/the/bar.baz", null));
 
     assertValidToken(result.sasToken(), BlobContainerSasPermission.parse("racwdl"), true);
   }
+
+  @Test
+  public void createAzureStorageContainerSasToken_basedOnLzSharedStorageAccount_Success()
```

Comment:
Is this named correctly? This test appears to be asserting that an IP address is added to the sas.

Reply:
Here we are testing token generation for storage container related to shared storage account. I used one of the parameters for token generation just for test purpose. So, to check that we still have correct token. It is possible to skip  all those parameters as well.